core/src/test/java/org/apache/iceberg/TestSchemaUpdate.java

```diff
@@ -1334,6 +1334,12 @@ public void testRemoveIdentifierFields() {
 
   @Test
   public void testSetIdentifierFieldsFails() {
+
+    AssertHelpers.assertThrows("Creating schema with optional identifier field should fail",
+        IllegalArgumentException.class,
+        "Cannot add field data as an identifier field: not a required field",
+        () -> new Schema(SCHEMA.asStruct().fields(), ImmutableSet.of(1, 2)));
+
```

Comment:
Is there a test for both double and float type already?

Reply:
added  test for float and double types